mdtraj/core/trajectory.py

```diff
@@ -460,29 +464,40 @@ def iterload(filename, chunk=100, **kwargs):
     if chunk == 0:
         # If chunk was 0 then we want to avoid filetype-specific code
         # in case of undefined behavior in various file parsers.
-        yield load(filename, **kwargs)
-
+        # TODO: this will first apply stride, then skip!
+        if extension not in _TOPOLOGY_EXTS:
+            kwargs['top'] = top
+        yield load(filename, **kwargs)[skip:]
     elif extension in ('.pdb', '.pdb.gz'):
         # the PDBTrajectortFile class doesn't follow the standard API. Fixing it
         # to support iterload could be worthwhile, but requires a deep refactor.
         t = load(filename, stride=stride, atom_indices=atom_indices)
         for i in range(0, len(t), chunk):
-            yield  t[i:i+chunk]
+            yield t[i:i+chunk]
 
     else:
         with (lambda x: open(x, n_atoms=topology.n_atoms)
               if extension in ('.crd', '.mdcrd')
               else open(filename))(filename) as f:
+            skipped = 0
```

Comment:
it would be more efficient here to just use the `seek` method on the open file handle.